server/services/SearchService.js

```diff
@@ -28,13 +28,33 @@ const searchByNick = (nickname) => {
   })
 };
 
-const findByTag = (val) => {
-  let postsRef = db.collection('posts');
-  return postsRef.where('tag', 'array-contains', val).get()
+const searchByTag = (tag) => {
+  return new Promise((resolve, reject) => {
```

Comment:
Why you need promise here? You alrdy have db.collection as promise

Reply:
I wrote searchByTag same as  searchByNick (as all http handlers in our code). There are lot of unnecessary promises. We can reactor all of that in separate pr.

Reply:
Let's for now refactor this one and in separate others.